### PR TITLE
Feat/language switcher

### PR DESCRIPTION
**Pull request by aurelien-brabant**
**Please consider merging the *feat-landing-testimonial-carrousel* branch before that one**

Add language switching capability through the `LanguageSwitcher` component. Language names are also translated.
Language change preserves scrolling position and uses client-side routing.

### CODE REVIEW

**Comment by aurelien-brabant** [comment]
I replaced the standard select with a tailwind UI select (see https://tailwindui.com/components/application-ui/forms/select-menus#component-c549ac2695455cb78d529c3a00293fe0)
Ready for merge @n1c01a5

File: hooks/useInterval.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { useEffect, useRef } from 'react'
+
+// Slightly modified version of overreacted's useInterval hook
+// See: https://overreacted.io/making-setinterval-declarative-with-react-hooks/
+
+export const useInterval = (callback: Function, delay: number) => {
```
[the remark this answers is not in the document]
Comment (aurelien-brabant) [reply hooks/useInterval.tsx:6]: looks like I need to fix the script again

**Comment by aurelien-brabant** [comment]
@n1c01a5  I saw the changes, will implement that ASAP

**Comment by n1c01a5** [comment]
Can you remove `yarn.lock` it's strange maybe your .gitignore is not updated.

**Comment by aurelien-brabant** [comment]
> Can you remove `yarn.lock` it's strange maybe your .gitignore is not updated.

If I recall correctly I asked if it was okay to remove it but you wanted to keep it since NextJS presents both lockfiles

**Comment by n1c01a5** [comment]
@aurelien-brabant I have trouble expressing myself. Both are present in the documentation but not in the example files (see https://github.com/vercel/next.js/tree/canary/examples/active-class-name), especially since it adds a warning about Vercel.

**Comment by aurelien-brabant** [comment]
@n1c01a5 proper layout has been implemented (select has a slightly different design but I guess it doesn't matter that much you can give a look a it)
yarn.lock as been deleted
Looking forward merge

**Pull request merged**